teps/0150-displayname-in-matrix.md

```diff
@@ -0,0 +1,304 @@
+      - name: browser
+        value: $(params.browsers[*])
+  taskRef:
+    name: run-tests
+```
+
+### `tasks[].matrix.displayName` and `tasks[].matrix.include[].name`
+
+`matrix.include[]` section allows specifying a `name` along with a list of `params`. The `name` field has no clear
+purpose in terms of utilization. We propose to further enhance the `name` field and make it available as part of the
+`pipelineRun.status.childReferences[].displayName` if specified. `spec.tasks[].matrix.displayName` and
+`spec.tasks[].matrix.include[].name` can co-exist but `spec.tasks[].matrix.include[].name` takes higher precedence. We
+propose `spec.tasks[].matrix.include[].name` is included in the `pipelineRun.status.childReferences` and
+`pipelineRun.status.childReferences[].displayName` is set to `spec.tasks[].matrix.include[].name`. Along with this
+proposal, the pipeline author can specify `params` in `spec.tasks[].matrix.include[].name` which are resolved in the
+`childReferences`.
+
+```yaml
+- name: platforms-and-browsers-with-include
+  matrix:
+    include:
+      - name: "Platform: $(params.platform)"
+        params:
+          - name: platform
+            value: linux111
+  params:
+    - name: browser
+      value: chrome
+- name: platforms-and-browsers-with-include-but-not-resolved
+  matrix:
+    include:
+      - name: "Platform: $(params.browser)"
+        params:
+          - name: platform
+            value: linux111
+  params:
+    - name: browser
+      value: chrome
+```
+
+### Precedence Order
+
+| specification                                                                            | precedence                      | `childReferences[].displayName` |
+|------------------------------------------------------------------------------------------|---------------------------------|---------------------------------|
+| `tasks[].displayName`                                                                    | `tasks[].displayName`           |                                 |
+| `tasks[].matrix.displayName`                                                             | `tasks[].matrix.displayName`    | `tasks[].matrix.displayName`    |
+| `tasks[].matrix.include[].name`                                                          | `tasks[].matrix.include[].name` | `tasks[].matrix.include[].name` |
+| `tasks[].displayName` and `tasks[].matrix.displayName`                                   | `tasks[].matrix.displayName`    | `tasks[].matrix.displayName`    |
+| `tasks[].displayName` and `tasks[].matrix.include[].name`                                | `tasks[].matrix.include[].name` | `tasks[].matrix.include[].name` |
+| `tasks[].matrix.displayName` and `tasks[].matrix.include[].name`                         | `tasks[].matrix.include[].name` | `tasks[].matrix.include[].name` |
+| `tasks[].displayName`, `tasks[].matrix.displayName`, and `tasks[].matrix.include[].name` | `tasks[].matrix.include[].name` | `tasks[].matrix.include[].name` |
```

Comment:
This is ok, but I wonder if only using `tasks[].displayName`  and introducing `matrix` specific variables wouldn't be more straight forward for users as well as implementation?